docs/DeveloperGuide.md

```diff
@@ -2,32 +2,285 @@
+  university name and contact type (email or phone number).
+* After parsing, the `findMatchingSchool()` method identifies the correct university entry within the JSON data.
+* The `handleContactType()` method retrieves and prints the requested contact information based on the input, 
+  displaying either the university’s email address or phone number.
+* There are also assertions and logging in place for error handling.
+
+#### Why it is implemented that way:
+* The `execute` method is essential and unique to every command class so inheritance was used.
+* Every method in the class remains maintainable and has one responsibility this allows easy debugging and
+  refactoring.
+* By using inheritance, new command classes can easily extend the functionality of existing ones 
+  which reducing redundancy in the code
+* Logging and assertions helps the team of developers to follow through the command execution.
+
+#### Alternatives considered:
+* Reading of the `database.json` was tricky and other libraries were considered.
+* Considered placing all the class methods inside the `execute` method but kept SLAP in mind to ensure
+  readability.
+
+#### Sequence Diagram on PlantUML:
+![Filter Courses Sequence Diagram](images/ObtainContactsCommand.png)
+
+### 4. List University Courses Command
+
+#### Overview
+
+This command is responsible for listing out all the mappable partner university’s (PU) courses and NUS courses. 
+This allows users to plan their course mapping as it lists out all the possible courses they can map in a specified
+partner university.
+
+#### How the feature is implemented:
+* The `ListUniCoursesCommand` class extends the `CheckInformationCommand` class where it overrides the execute method 
+for custom behaviour.
+* The command first reads a JSON file to obtain the JsonObject containing the names of all the partner universities.
+* The getPuName method then parses and extracts the PU name from the input (with the format of `set [PU_NAME]`).
+* Next, the extracted PU name is passed into the `getUniCourses()` method which will search for the specified PU in the
+JsonObject with `findUniversityName()`.
+* If the university is not found, an exception `UnknownUniversityException` will be thrown.
+* If the PU is found, the `listCourses()` will be called. Then `getUniversityObject()` and `getCourseArray()` methods
+will be called to get the JsonObject containing the PU and the JsonArray containing the list of courses it offers.
+* The two object will be passed into `iterateCourses()` method to iterate through the JsonArray `courseArray` which
+contains the list of courses.
+* It then prints out the course details such as PU course code and NUS course code through the `printCourseDetails()`
+method.
+* Assertions and logging are used for error handling.
+
+#### Sequence on PlantUML:
+![ListUniCourseCommand sequence diagram](images/ListUniCoursesCommand.png)
```

Comment:
Should there be an activation bar and dotted return arrow for "print course details"?